### PR TITLE
add smolvla to model-libraries.ts

### PR DESCRIPTION
**Pull request by danaaubakirova**

### CODE REVIEW

File: packages/tasks/src/model-libraries-snippets.ts [hunk 1]
```diff
@@ -1637,4 +1637,29 @@ audio = model.autoencoder.decode(codes)[0].cpu()
...
+  --policy.path=lerobot/smolvla_base \\
+  --dataset.repo_id=lerobot/svla_so101_pickplace \\
+  --batch_size=64 \\
+  --steps=20000 \\
+  --output_dir=outputs/train/my_smolvla \\
+  --job_name=my_smolvla_training \\
+  --policy.device=cuda \\
+  --wandb.enable=true`,
+];
+
```
Comment (pcuenca) [inline packages/tasks/src/model-libraries-snippets.ts]: We usually demonstrate simple inference snippets. Does that make sense for smolvla, or not really?

File: packages/tasks/src/model-libraries.ts [hunk 2]
```diff
@@ -1103,6 +1103,14 @@ export const MODEL_LIBRARIES_UI_ELEMENTS = {
 		countDownloads: `path:"model_vae_fp16.pt"`,
 		snippets: snippets.threedtopia_xl,
 	},
+	smolvla: {
+		prettyLabel: "SmolVLA",
+		repoName: "🤗/lerobot",
+		repoUrl: "https://github.com/huggingface/lerobot/blob/main/lerobot/common/policies/smolvla",
```
Comment (pcuenca) [inline packages/tasks/src/model-libraries.ts]:
```suggestion
		repoUrl: "https://github.com/huggingface/lerobot",
```

File: packages/tasks/src/model-libraries-snippets.ts [hunk 3]
```diff
@@ -1637,4 +1637,29 @@ audio = model.autoencoder.decode(codes)[0].cpu()
...
+	`# git clone https://github.com/huggingface/lerobot.git
+cd lerobot
+conda create -y -n lerobot python=3.10
+conda activate lerobot
+
+# Install ffmpeg (required for video processing)
+conda install ffmpeg=7.1.1 -c conda-forge
+
+# Install LeRobot with the SmolVLA extra dependencies
+pip install -e ".[${model.id}]"
```
Comment (Wauplin) [inline packages/tasks/src/model-libraries-snippets.ts]:
```suggestion
pip install -e ".[smolvla]"
```

having a dynamic extra dependency feels off. Shouldn't it be `".[smolvla]"` instead?

Comment (Wauplin) [inline packages/tasks/src/model-libraries-snippets.ts]: I feel that all of this should be simplified to:

```suggestion
	`# !pip install "git+https://github.com/huggingface/lerobot.git#egg=lerobot[smolvla]"
```

i.e. no need for editable mode, no need to download the repo just to install it, no need for conda instructions (users can figure that part by themselves, depending on what they prefer).

If users want more detailed installation guide, they can go to https://github.com/huggingface/lerobot?#installation

File: packages/tasks/src/model-libraries-snippets.ts (hunk 1 above)
Comment (Wauplin) [reply packages/tasks/src/model-libraries-snippets.ts]: agree with @pcuenca that an inference example feels more natural at this stage since if a user has pushed their model, it means it has already been trained (at least I expect). 

FYI, `export const smolvla = (model: ModelData): string[] => [` returns a list so it could return both a training and an inference examples.

File: packages/tasks/src/model-libraries-snippets.ts [hunk 4]
```diff
@@ -1637,4 +1637,29 @@ audio = model.autoencoder.decode(codes)[0].cpu()
...
+
+# Launch finetuning on your dataset
+cd lerobot && python lerobot/scripts/train.py \\
+  --policy.path=lerobot/smolvla_base \\
```
Comment (Wauplin) [inline packages/tasks/src/model-libraries-snippets.ts]:
```suggestion
  --policy.path=${model.id} \\
```

if we really want to showcase the training script, it should load the model from the repo as base policy isn't it?

File: packages/tasks/src/model-libraries.ts [hunk 5]
```diff
@@ -1103,6 +1103,14 @@ export const MODEL_LIBRARIES_UI_ELEMENTS = {
 		countDownloads: `path:"model_vae_fp16.pt"`,
 		snippets: snippets.threedtopia_xl,
 	},
+	smolvla: {
+		prettyLabel: "SmolVLA",
+		repoName: "🤗/lerobot",
```
Comment (Wauplin) [inline packages/tasks/src/model-libraries.ts]:
```suggestion
		repoName: "LeRobot",
```

(I know we use the emoji for transformers/diffusers but I'm not really a fan of it...)

**Review by Wauplin — COMMENTED** [review]
Left a few comments. Agree on the importance of havign an inference example (or both train and inference if that's more natural).

Also I'm wondering if it shouldn't be `lerobot` entirely instead of `smolvla`. Is `smolvla` really considered to be a library on its own or it is more a subpart of the library (e.g. LeRobot has many policies as transformers have many architectures). 
Using correct tags e.g.
```yaml
library_name: lerobot
tags:
    - smolvla
```
we should be able to display high-level information related to the library (i.e. repo name, pretty label, repoUrl, docsUrl, etc.) while having dedicated snippets for each policy (`export const smolvla = (model: ModelData): string[] => [` is a method and we have access to the model tags so we can do interesting stuff there).

File: packages/tasks/src/model-libraries-snippets.ts (hunk 1 above)
Comment (danaaubakirova) [reply packages/tasks/src/model-libraries-snippets.ts]: Yes, I see what you mean. However, unfortunately we cannot use smolvla_base for inference directly, the only use case for smolvla_base for now is fine-tuning. 
As it is not meant to perform zero-shot and will cause many errors. Hence, we strongly recommend people fine-tuning the model first.

We could have pushed the inference snippet for any model fine-tuned on top of smolvla_base however, but that is not really applicable here I think. wdyt?

File: packages/tasks/src/model-libraries-snippets.ts (hunk 3 above)
Comment (danaaubakirova) [reply packages/tasks/src/model-libraries-snippets.ts]: nice! thanks!

**Comment by danaaubakirova** [comment]
Thanks for the review! @Wauplin I changed the tag to `lerobot` in model-libraries following your reasoning about smolvla being the tag and lerobot being main library. Lmk if this looks good!

File: packages/tasks/src/model-libraries-snippets.ts (hunk 1 above)
Comment (Wauplin) [reply packages/tasks/src/model-libraries-snippets.ts]: _(originally posted in DMs on slack)_

> However, unfortunately we cannot use smolvla_base for inference directly, the only use case for smolvla_base for now is fine-tuning.
Interesting! What I would suggest:
- if `modelId is smolvla_base` => then provide the finetune command
- if `modelId != smolvla_base` => then provide the "how to use" command + the finetune command

Since `export const smolvla = (model: ModelData): string[] => [` is a method and you have access to `ModelData` you can easily add some simple logic like this to the snippet generation

I see there are 380+ models on https://huggingface.co/models?other=lerobot and only 1 is not finetuned (if I understand correctly?) so it would be a shame not to show the inference command for all of them. What do you think?

File: packages/tasks/src/model-libraries.ts [hunk 6]
```diff
@@ -527,6 +527,14 @@ export const MODEL_LIBRARIES_UI_ELEMENTS = {
 		filter: false,
 		countDownloads: `path_extension:"litertlm"`,
 	},
+	lerobot: {
+		prettyLabel: "SmolVLA",
+		repoName: "LeRobot",
+		repoUrl: "https://github.com/huggingface/lerobot",
+		docsUrl: "https://huggingface.co/docs/lerobot/smolvla",
+		filter: false,
+		snippets: snippets.smolvla,
```
Comment (Wauplin) [inline packages/tasks/src/model-libraries.ts]:
```suggestion
		snippets: snippets.lerobot,
```

File: packages/tasks/src/model-libraries.ts [hunk 7]
```diff
@@ -527,6 +527,14 @@ export const MODEL_LIBRARIES_UI_ELEMENTS = {
 		filter: false,
 		countDownloads: `path_extension:"litertlm"`,
 	},
+	lerobot: {
+		prettyLabel: "SmolVLA",
```
Comment (Wauplin) [inline packages/tasks/src/model-libraries.ts]:
```suggestion
		prettyLabel: "LeRobot",
```

File: packages/tasks/src/model-libraries.ts [hunk 8]
```diff
@@ -527,6 +527,14 @@ export const MODEL_LIBRARIES_UI_ELEMENTS = {
...
+		repoName: "LeRobot",
+		repoUrl: "https://github.com/huggingface/lerobot",
+		docsUrl: "https://huggingface.co/docs/lerobot/smolvla",
```
Comment (Wauplin) [inline packages/tasks/src/model-libraries.ts]:
```suggestion
		docsUrl: "https://huggingface.co/docs/lerobot",
```

File: packages/tasks/src/model-libraries-snippets.ts [hunk 9]
```diff
@@ -1735,4 +1735,20 @@ audio = model.autoencoder.decode(codes)[0].cpu()
...
+cd lerobot && python lerobot/scripts/train.py \\
+  --policy.path=${model.id} \\
+  --dataset.repo_id=lerobot/svla_so101_pickplace \\
+  --batch_size=64 \\
+  --steps=20000 \\
+  --output_dir=outputs/train/my_smolvla \\
+  --job_name=my_smolvla_training \\
+  --policy.device=cuda \\
+  --wandb.enable=true`,
+];
```
Comment (Wauplin) [inline packages/tasks/src/model-libraries-snippets.ts]: Here is logic for `smolvla` models. You will be able to extend this logic to other policies (e.g. `act`?) in a future PR.

```suggestion
export const lerobot = (model: ModelData): string[] => {
	if (model.tags.includes("smolvla")) {
		const smolvlaSnippets = [
			// Installation snippet
			`# Install LeRobot from source
git clone https://github.com/huggingface/lerobot.git
cd lerobot
pip install -e .[smolvla]`,
			// Finetune snippet
			`# Launch finetuning on your dataset
python lerobot/scripts/train.py \\
--policy.path=${model.id} \\
--dataset.repo_id=lerobot/svla_so101_pickplace \\
--batch_size=64 \\
--steps=20000 \\
--output_dir=outputs/train/my_smolvla \\
--job_name=my_smolvla_training \\
--policy.device=cuda \\
--wandb.enable=true`,
		];
		if (model.id !== "lerobot/smolvla_base") {
			// Inference snippet (only if not base model)
			smolvlaSnippets.push(
				`# Run the policy
<TODO add inference snippet here>`
			);
		}
		return smolvlaSnippets;
	}
	return [];
};
```

Comment (Wauplin) [reply packages/tasks/src/model-libraries-snippets.ts]: > git clone https://github.com/huggingface/lerobot.git

(nit: would be good if it was possible to launch the scripts without cloning LeRobot repo i.e. simply after a `pip install lerobot`)

Comment (Wauplin) [reply packages/tasks/src/model-libraries-snippets.ts]: (nit) move the `lerobot` method in `packages/tasks/src/model-libraries-snippets.ts` to preserve alphabetical order

**Review by Wauplin — COMMENTED** [review]
As discussed in DMs, here is my suggestion for LeRobot snippets based on tags. For this to work, all smolvla models must include `tags: - smolvla` in their model card metadata. Doing this will make them listed here https://huggingface.co/models?other=smolvla. This is typically something lerobot library should do automatically to that users don't have to do it themselves. Updating lerobot library will benefit all future models. For existing ones, you might want to open PRs on the Hub to correct them (similar to https://huggingface.co/lerobot/smolvla_base/discussions/7).

File: packages/tasks/src/model-libraries-snippets.ts [hunk 10]
```diff
@@ -1735,4 +1735,46 @@ audio = model.autoencoder.decode(codes)[0].cpu()
 torchaudio.save("sample.wav", audio, model.autoencoder.sampling_rate)
 `,
 ];
+
+export const lerobot = (model: ModelData): string[] => {
```
Comment (Wauplin) [inline packages/tasks/src/model-libraries-snippets.ts]: (nit) can you move it up in the file for alphabetical order?

**Review by Wauplin — APPROVED** [review]
Thanks for updating @danaaubakirova! Except a small nit, PR looks good to merge now. Pinging @pcuenca @Vaibhavs10 if you'd like to have a quick look. 

@danaaubakirova as I mentioned above, in parallel of this PR it's very important for the LeRobot library to include more metadata when generating new models (library_name, tags, base_model, citation, etc.). A tighter integration will benefit users on the long run. If you need any help or advice about this, please let me know :hugs:  

---

**Note:** CI failure is unrelated

File: packages/tasks/src/model-libraries-snippets.ts (hunk 10 above)
Comment (danaaubakirova) [reply packages/tasks/src/model-libraries-snippets.ts]: done! thanks a lot!

**Comment by danaaubakirova** [comment]
> Thanks for updating @danaaubakirova! Except a small nit, PR looks good to merge now. Pinging @pcuenca @Vaibhavs10 if you'd like to have a quick look.
> 
> @danaaubakirova as I mentioned above, in parallel of this PR it's very important for the LeRobot library to include more metadata when generating new models (library_name, tags, base_model, citation, etc.). A tighter integration will benefit users on the long run. If you need any help or advice about this, please let me know 🤗
> 
> **Note:** CI failure is unrelated

TYSM @Wauplin! Sure, we can work on generating more informative and well-organised metadata when pushing models using lerobot, and keep you in the loop, in case we will have any questions:)

File: packages/tasks/src/model-libraries-snippets.ts [hunk 11]
```diff
@@ -1735,4 +1776,6 @@ audio = model.autoencoder.decode(codes)[0].cpu()
 torchaudio.save("sample.wav", audio, model.autoencoder.sampling_rate)
 `,
 ];
+
+
```
Comment (pcuenca) [inline packages/tasks/src/model-libraries-snippets.ts]:
```suggestion

```

**Review by pcuenca — APPROVED** [review]
Great suggestions @Wauplin, thanks a lot!

We could potentially:
- Use a simplified install command (as @Wauplin mentioned), plus the inference snippet if the model is not the base one.
- Use the more complicated install + fine-tune example when it is.

Currently, IIUC we are showing all the snippets if the model is not the base one, is that what we intended?

Happy to merge after Wauplin's approval anyway.

**Comment by Wauplin** [comment]
> We could potentially:
> - Use a simplified install command (as @Wauplin mentioned), plus the inference snippet if the model is not the base one.
> - Use the more complicated install + fine-tune example when it is.
> Currently, IIUC we are showing all the snippets if the model is not the base one, is that what we intended?

I will leave the final decision to @danaaubakirova! In my understanding, users could want to retrain/finetune an already finetuned model but maybe that's not the main use case? In any case, both are fine for me. @danaaubakirova let us know what you think!

**Comment by danaaubakirova** [comment]
Hi @pcuenca, thanks for the fix!

- The simplified install command `pip install "git+https://github.com/huggingface/lerobot.git#egg=lerobot[smolvla]"`
doesn’t handle our environment setup. We still need a specific ffmpeg version installed via conda, so I’d point people to the install guide rather than rely on the one-liner or at least setup conda environment before the installation.

- Right now we “force” everyone to fine-tune smolvla_base since that’s the only supported workflow. For completed, fine-tuned models we are shipping inference snippets, but users who really want to re-fine-tune those models can still do so as noted by @Wauplin. 
 
 Let me know if I can improve this PR further given the constraints I described above :)
 
P.S. One downside of user-uploaded, fine-tuned models is that they’re tied to specific datasets and don’t generalize well, so they can’t be freely reused across different setups/users.

File: packages/tasks/src/model-libraries-snippets.ts [hunk 12]
```diff
@@ -762,6 +762,47 @@ print(output)`);
 	return snippets;
 };
 
+export const lerobot = (model: ModelData): string[] => {
+	if (model.tags.includes("smolvla")) {
+		const smolvlaSnippets = [
+			// Installation snippet
+			`# Install LeRobot from source
```
Comment (Wauplin) [inline packages/tasks/src/model-libraries-snippets.ts]:
```suggestion
			`# See https://github.com/huggingface/lerobot?tab=readme-ov-file#installation for more details
```

**Comment by Wauplin** [comment]
Let's merge it "as-is" then and we can reevaluate later. Looks like the most important for now is to have correct snippets for https://huggingface.co/lerobot/smolvla_base + have a "Use with LeRobot" button in all others.
